### ISSUE TITLE
Plotting performance of different datasets with the same size

### ISSUE THREAD

**Issue by PPNav** [oxyplot/oxyplot#757]
I simulated two big datasets. Each dataset has 50000 points. The first one is (x=1:50000, y= 1:50000, step = 1), and the second one is (x = 1:50000, y=Math.Sin(i), where i=1:50000, step = 1). Then I plotted the two datasets respectively, I found the velocity of plotting the first dataset is much faster than that of the second one. I want to know what makes the significant difference? I am confusing now.

**Comment by objorke** [comment]
I agree this should not be very different since both curves are continuous. Or was the first curve simplified to two points?
But expect lower performance if you have y-values like 0,1,0,1,0,1,... (in the example browser)

**Comment by PPNav** [comment]
[Uploading OxyPlotTest.zip…]()

**Comment by PPNav** [comment]
[Uploading OxyPlotTest0.zip…]()

In my program, I want to use oxyplot to plot millions of data points, so I am sensitive to its performance.

I also found that plotting scatteseries is much faster than plotting lineseries. So, I think that perhaps LineJoin is a time-consuming work when plotting lines!

Please review and check my source code in Form_Main.cs. And, the OxyPlotTest is builded under VS2013 and .net framework 4.5.

Best Regards.

I am looking forward your reply.

**Comment by PPNav** [comment]
[Form_Main.zip](https://github.com/oxyplot/oxyplot/files/115289/Form_Main.zip)
[OxyPlotTest0.zip](https://github.com/oxyplot/oxyplot/files/115290/OxyPlotTest0.zip)

Here are my source codes.

**Comment by objorke** [comment]
thanks for including code!
Even if the number of points is the same, the number of pixels that must be drawn is very different.
Could this explain the issue?
